lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -726,14 +790,35 @@ public ScorerSupplier scorerSupplier(LeafReaderContext context) throws IOExcepti
       DocIdSet docIdSet;
       try {
         docIdSet = get(in.getQuery(), cacheHelper);
+      } catch (AsyncQueryLoadInProgressException e) {
```

Comment:
We could alternatively have this request wait for cache to populate and then return that cached result.  It'd be cheaper CPU overall, especially in degenerate queries where the same query is often executed multiple times concurrently, but I think more complicated here, so let's go forward w/ this approach?

Reply:
+1, I agree